retrofit/src/main/java/retrofit/MethodHandler.java

```diff
@@ -28,7 +29,9 @@
     Converter<ResponseBody, Object> responseConverter =
         (Converter<ResponseBody, Object>) createResponseConverter(method, retrofit, responseType);
     RequestFactory requestFactory = RequestFactoryParser.parse(method, responseType, retrofit);
-    return new MethodHandler<>(retrofit, requestFactory, callAdapter, responseConverter);
+    String infoForException = collectInfoForException(method);
```

Comment:
Yikes. Is there any way to make this lazily computed when needed? I realize we're caching it so the cost is only paid once, but since it's in an exceptional case I don't mind paying the cost of computation every time.